miden/src/cli/prove.rs

```diff
@@ -61,9 +53,24 @@ impl ProveCmd {
         let advice_provider = input_data.parse_advice_provider()?;
 
         // execute program and generate proof
-        let (stack_outputs, proof) =
-            prover::prove(&program, stack_inputs, advice_provider, &self.get_proof_security())
-                .map_err(|err| format!("Failed to prove program - {:?}", err))?;
+        let proof = match self.security.as_str() {
+            "96bits" => prover::prove(
```

Comment:
maybe use [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html)?

Reply:
That is something to think about. This is not really a property of the proof options, but instead it is a CLI argument that we accept. I wonder if we should couple the two